frontend/public/components/catalog/catalog-items.jsx

```diff
@@ -76,387 +131,305 @@ export class CatalogTileViewPage extends React.Component {
+    const { categories, selectedCategoryId } = this.state;
+    const { subcategories } = category;
 
-    if (byKeyword.active) {
-      return this.filterByKeyword(byKeyword.value, items);
+    if (selectedCategoryId === 'all') {
+      return _.map(categories, topCategory => {
+        if (topCategory.id === 'all') {
+          return null;
+        }
+        return this.renderCategory(topCategory, false);
+      });
     }
 
-    return items;
-  }
-
-  clearFilters() {
-    const params = new URLSearchParams(window.location.search);
-    params.delete(KEYWORD_URL_PARAM);
-    params.delete(TYPE_URL_PARAM);
-    const url = new URL(window.location);
-    const searchParams = `?${params.toString()}${url.hash}`;
-    history.replace(`${url.pathname}${searchParams}`);
-    if (this.unmounted) {
-      return;
+    if (!_.size(subcategories)) {
+      return this.renderCategory(category, true);
     }
-    this.setState(this.getUpdatedStateFromURLParams(searchParams));
-    this.filterByKeywordInput.focus();
-  }
 
-  updateURL(filterName, value) {
-    const params = new URLSearchParams(window.location.search);
-    if (value) {
-      params.set(filterName, Array.isArray(value) ? JSON.stringify(value) : value);
-    } else {
-      params.delete(filterName);
-    }
-    const url = new URL(window.location);
-    const searchParams = `?${params.toString()}${url.hash}`;
-    history.replace(`${url.pathname}${searchParams}`);
-    // do not set state if unmounted, case for when history.replace taking too much time
-    // and causes component to remount.  Noticeable in Safari
-    if (this.unmounted) {
-      return;
-    }
-    this.setState(this.getUpdatedStateFromURLParams(searchParams));
+    return _.map(subcategories, subcategory => this.renderCategory(subcategory, false));
   }
 
-  onFilterChange(filterType, id, value) {
-    const { filters } = this.state;
-    const { byType } = filters;
-    let byTypeValues = [];
-
-    if (filterType === 'byKeyword') {
-      this.updateURL(KEYWORD_URL_PARAM, value);
-    } else {
-      if (value) {
-        byTypeValues.push(id);
-      }
-      _.forOwn(byType, (typeKeyValue, typeKey) => {
-        if (typeKey !== id && byType[typeKey].active) {
-          byTypeValues.push(typeKey);
-        }
-      });
-      this.updateURL(TYPE_URL_PARAM, _.isEmpty(byTypeValues) ? false : byTypeValues);
-    }
+  renderEmptyState() {
+    return (
+      <EmptyState className="co-catalog-page__no-filter-results">
+        <EmptyState.Title className="co-catalog-page__no-filter-results-title" aria-level="2">
+          No Results Match the Filter Criteria
+        </EmptyState.Title>
+        <EmptyState.Info className="text-secondary">
+          No catalog items are being shown due to the filters being applied.
+        </EmptyState.Info>
+        <EmptyState.Help>
+          <button type="text" className="btn btn-link" onClick={() => this.clearFilters()}>Clear All Filters</button>
```

Comment:
Bind clear filters to this

```ts
clearFilters = () => {
  // ...
};
```

Then you don't need the arrow function here:

```suggestion
          <button type="text" className="btn btn-link" onClick={this.clearFilters}>Clear All Filters</button>
```